helix-lock/src/main/java/org/apache/helix/lock/HelixLockScope.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.helix.lock;
+
+import java.util.List;
+
+import org.apache.helix.util.StringTemplate;
+
+
+/**
+ *  Defines the various scopes of Helix locks, and how they are represented on Zookeeper
+ */
+public class HelixLockScope {
```

Comment:
It could be cleaner if we have an abstract HelixScope class. Then there should be no duplicate code at all.
Then things we can put into the parent class:
- StringTemplate template
- clusterName
- zkPath
- type
- their get methods
- participant name is also something in common. But I think this one is not generic enough for the parent abstract class.

Reply:
You are right, it would be cleaner if we have an abstract class. But I decided not to do it at this time because 
1. HelixLockScope and HelixConfigScope don't share the same logic for most part, for example: HelixConfigScope takes a mapKey as a variable in the constructor while HelixLockScope does not, and these two classes don't define same zkPath argument number for each level so the getZkPath logic is different too. 
2. We want to decouple Helix Lock module from Helix core module, so in the future if needed we can freely extend the lock scope for other usages of the lock module.

Reply:
Discussed offline. Will make this LockScope more generic so user can define the properties themselves.